base/file.jl

```diff
@@ -571,8 +605,8 @@ end
 Apply the function `f` to the result of [`mktemp(parent)`](@ref) and remove the
 temporary file upon completion.
 """
-function mktemp(fn::Function, parent=tempdir())
-    (tmp_path, tmp_io) = mktemp(parent)
+function mktemp(fn::Function, parent::AbstractString=tempdir())
+    (tmp_path, tmp_io) = mktemp(parent, cleanup=false)
```

Comment:
Should we make this `cleanup=true`? That way, if for some reason the `rm(tmp_path)` inside the `finally` block fails, we'll have a second chance at deleting the file when Julia exits.

Reply:
I guess, I would prefer not to add more work for the finalizer if we don't have to though. I suppose we could check if the path still exists after trying to remove it and add it then.

Reply:
That sounds good. We'd only need to add the line
```julia
ispath(tmp_path) && temp_cleanup_list_push!(tmp_path)
```
between line 620 and line 621.

Reply:
You beat me to it.

Reply:
Ok, I changed it so that if the removal fails, then we add the path to the cleanup list so that if the path can be removed at exit it will be.

Reply:
And now I further changed it so that we keep trying to delete paths that should already have been deleted.